.github/scripts/update-copyright-year.py

```
@@ -0,0 +1,156 @@
+        return False, []
+    
+    try:
+        content = file_path.read_text(encoding='utf-8')
+        original_content = content
+        changes = []
+        
+        for pattern, replacement in PATTERNS:
+            if callable(replacement):
+                # For date range patterns, use lambda function
+                def replacer(match):
+                    result = replacement(match)
```

Ruff (0.14.10)

Function definition does not bind loop variable `replacement`

(B023)